Support latest windows image 1.2.8  with 3ds Max 2019

matthew-rollitt opened the pull request:
Updated templates to handle the new windows version 1.2.8 which includes 3dsmax 2019 and refactored some templates since we can use configuration variables now.

matthew-rollitt committed:
Updated templates to handle the new windows version 1.2.8 which includes 3dsmax 2019 and refactored some templates since we can use configuration variables

matthew-rollitt committed:
Merge remote-tracking branch 'origin/master' into support-latest-image

ascobie commented:
Seems ok, but need Christian to look over it as well.

ascobie commented on ncj/3dsmax/vray-dr/job.template.json:
```diff
@@ -199,7 +210,7 @@
                             "numberOfInstances":"[parameters('nodeCount')]",
-                            "coordinationCommandLine":"cmd.exe /c if \"[parameters('vrayRenderer')]\" == \"VRayAdv\" (cmd.exe /c start \"\" \"C:\\Autodesk\\3dsMax2018\\vrayspawner2018.exe\" \"-port=[parameters('vrayPort')]\") else (cmd.exe /c start C:\\server\\3dsmaxVray\\bin\\vray.exe -server -portNumber=[parameters('vrayPort')])",
+                            "coordinationCommandLine":"cmd.exe /c if (\"[parameters('vrayRenderer')]\" == \"VRayAdv\") if (\"[parameters('maxVersion')]\" == \"3ds Max 2018\") cmd.exe /c start \"\" \"C:\\Autodesk\\3dsMax2018\\vrayspawner2018.exe\" \"-port=[parameters('vrayPort')]\" if (\"[parameters('maxVersion')]\" == \"3ds Max 2019\")   cmd.exe /c start \"\" \"C:\\Autodesk\\3dsMax2019\\vrayspawn ...
                             "commonResourceFiles":[
```
coordinationCommandLine in the DR template is impossible to make sense of. Can we pull this out into another script to make it easier to read?

matthew-rollitt replied:
I think it's possible to pull them out.

matthew-rollitt committed:
replaced coordinationCommandLine

replaced coordinationCommandLine with a bat file and updated
https://raw.githubusercontent.com/Azure/BatchLabs-data/ to
https://raw.githubusercontent.com/Azure/BatchExplorer-data

matthew-rollitt committed:
Merge branch 'master' into support-latest-image

smith1511 commented on ncj/3dsmax/scripts/3dsmax.ps1:
```diff
@@ -209,17 +210,34 @@ mkdir -Force images > $null
 {
-    $max_exec = $env:3DSMAX_2018_EXEC
+        $max_exec = $env:3DSMAX_2019_EXEC
+        if(![System.IO.File]::Exists($max_exec))
```
Would be good to be consistent.

smith1511 commented on ncj/3dsmax/scripts/3dsmax.ps1:
```diff
@@ -209,17 +210,34 @@ mkdir -Force images > $null
-
+Else 
+{
+    Write-Host "No version of 3ds Max was selected."
```
Default to latest?  2019

smith1511 commented on ncj/3dsmax/scripts/PrepareVrayServices.bat:
```diff
@@ -0,0 +1,19 @@
+@echo off
+IF %4=="VRayAdv" (
```
Would be good to assign variables at the top for clarity.

set VRAY_RENDERER=%4

smith1511 replied:
Or use powershell

smith1511 commented on ncj/3dsmax/standard/job.parameters.json:
```diff
@@ -17,6 +17,9 @@
       "value":"MyScene.mxp"
    },
+   "maxVersion":{
+      "value":"3ds Max 2019"
```
Remove prefix?  Just "2019"?

smith1511 commented on ncj/3dsmax/scripts/3dsmax.ps1:
```diff
@@ -209,17 +210,34 @@ mkdir -Force images > $null
 # Render
 $max_exec = "3dsmaxcmdio.exe"
-if ($env:3DSMAX_2018 -and (Test-Path "$env:3DSMAX_2018"))
+If ($maxVersion -eq "3ds Max 2018")
```
Just version

smith1511 commented on ncj/3dsmax/standard/job.template.json:
```diff
@@ -18,6 +18,17 @@
+                "description": "The renderer to use"
+            },
+            "allowedValues":[
+                "3ds Max 2018",
```
Just version

smith1511 commented on ncj/3dsmax/vray-dr/job.parameters.json:
```diff
@@ -11,6 +11,9 @@
         "value":""
     },
+    "maxVersion":{
+      "value":"3ds Max 2019"
```
Just version

smith1511 commented on ncj/3dsmax/vray-dr/job.template.json:
```diff
@@ -18,6 +18,17 @@
         },
+        "maxVersion": {
+            "type": "string",
+            "defaultValue": "3ds Max 2019",
```
Just version

smith1511 commented on ncj/3dsmax/vray-dr/job.template.json:
```diff
@@ -199,14 +216,18 @@
                         "multiInstanceSettings":{
                             "numberOfInstances":"[parameters('nodeCount')]",
-                            "coordinationCommandLine":"cmd.exe /c if \"[parameters('vrayRenderer')]\" == \"VRayAdv\" (cmd.exe /c start \"\" \"C:\\Autodesk\\3dsMax2018\\vrayspawner2018.exe\" \"-port=[parameters('vrayPort')]\") else (cmd.exe /c start C:\\server\\3dsmaxVray\\bin\\vray.exe -server -portNumber=[parameters('vrayPort')])",
+                            "coordinationCommandLine":"cmd /c %AZ_BATCH_JOB_PREP_WORKING_DIR%/PrepareVrayServices.bat --MaxVersion [parameters('maxVersion')] --VrayRenderer [parameters('vrayRenderer')] --VrayPort [parameters('vrayPort')])",
```
Don't use these named args as you don't check for them in the batch file.  Just be clear in the batch file that the args are: <MaxVersion> <VRayRenderer> <VRayPort>

matthew-rollitt committed:
Updated templates based on a CR

Removed *3ds max* from the templates since you only need the year number.
Updated the bat file to make it easier to read and removed the powershell way of handling the args

matthew-rollitt committed:
Updated 1.2.8 to 1.2.9

Updated 1.2.8 to 1.2.9 for all the rendering windows images. Small update to the rendering images. The version increase was do up a version bump of arnold.

smith1511 commented on ncj/3dsmax/vray-dr/job.template.json:
```diff
@@ -199,14 +216,18 @@
                         "multiInstanceSettings":{
                             "numberOfInstances":"[parameters('nodeCount')]",
-                            "coordinationCommandLine":"cmd.exe /c if \"[parameters('vrayRenderer')]\" == \"VRayAdv\" (cmd.exe /c start \"\" \"C:\\Autodesk\\3dsMax2018\\vrayspawner2018.exe\" \"-port=[parameters('vrayPort')]\") else (cmd.exe /c start C:\\server\\3dsmaxVray\\bin\\vray.exe -server -portNumber=[parameters('vrayPort')])",
+                            "coordinationCommandLine":"cmd /c %AZ_BATCH_JOB_PREP_WORKING_DIR%/PrepareVrayServices.bat [parameters('maxVersion')] [parameters('vrayRenderer')] [parameters('vrayPort')])",
```
These args aren't in the correct order.  The script takes <VRay> <Max> <Port>.  I assume this hasn't been tested, can you test the changes before you merge please?

matthew-rollitt replied:
Updated the bat file 👍

matthew-rollitt committed:
Fixed the bat file

matthew-rollitt committed:
Reverted back the link to some resource files

Reverted back the link to some resource files

matthew-rollitt committed:
Merge branch 'master' into support-latest-image

matthew-rollitt committed:
Merge branch 'master' into support-latest-image

matthew-rollitt committed:
update render-windows-frames pool version
